tftb/generators/misc.py

```diff
@@ -174,58 +185,72 @@ def mexhat(nu=0.05):
     return h
 
 
-def gdpower(n_points, degree=0, rate=1):
+def gdpower(n_points, degree=0.0, rate=1.0):
     """Generate a signal with a power law group delay.
 
     :param n_points: Number of points in time.
     :param degree: degree of the power law.
     :param rate: rate-coefficient of the power law GD.
     :type n_points: int
-    :type degree: int
+    :type degree: int		# yoder: i'm pretty sure this is supposed to be a float. at least in test_misc.py, the test for this funct. passes degree=0.5
     :type rate: float
     :return: Tuple of time row containing modulated samples, group delay, \
             frequency bins.
     :rtype: tuple
     """
+    # quickly, handle types:
+    n_points = int(n_points)
+    degree   = float(degree)
+    rate     = float(rate)
+    #print("diagnostic: running gdpower")
     t0 = 0
-    lnu = np.round(n_points / 2)
+    #lnu = int(np.round(n_points / 2))
+    lnu = int(np.ceil(n_points / 2))		# ??
     nu = np.linspace(0, 0.5, lnu + 1)
     nu = nu[1:]
-    am = nu ** ((degree - 2) / 6)
+    #am = nu ** ((degree - 2) / 6)
+    am = nu ** ((degree - 2.0) / 6.0)
 
-    if rate == 0:
+    if rate == 0.:
         raise TypeError("rate must be non-zero")
 
     tfx = np.zeros((n_points,), dtype=complex)
 
-    if (degree < 1) and (degree != 0):
-        d = n_points ** (degree * rate)
+    if (degree < 1.) and (degree != 0):
+        #d = n_points ** (degree * rate)
+        d = float(n_points) ** (degree * rate)
         t0 = n_points / 10.0
-        tfx[:lnu] = np.exp(-1j * 2 * pi * (t0 * nu + d * nu ** degree / degree)) * am
+        #tfx[:lnu] = np.exp(-1j * 2 * pi * (t0 * nu + d * nu ** degree / degree)) * am
+        tfx[:lnu] = np.exp(-1.0j * 2. * pi * ((t0 * nu + d * nu ** degree) / float(degree))) * am
```

Comment:
agreed, but i often like to explicitly cast my floats 1) in case something changes in a later revision of the interpreter/compiler, 2) to communicate that the float type is important, and 3) just in case a later revision introduces a grouping that produces a problem, akin to ( int(1)/int(2) )*2.0 ?=  2.0*int(1)/int(2). but that sort of communication can be limited to the code review process as well; your decision.